### PR TITLE
update new unittests of flatten ops and layernorm, *test=kunlun

### PR DESCRIPTION
**Pull request by ykkk2333**
### PR types
Others 

### PR changes
Others 

### Describe
update unittests of flatten ops and layernorm to new style, *test = kunlun

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/xpu/test_flatten2_op_xpu.py [hunk 1]
```diff
@@ -23,67 +23,77 @@
...
 
-    def test_check_output(self):
-        self.check_output_with_place(self.place, no_check_set=["XShape"])
+        def setUp(self):
+            self.set_xpu()
+            self.op_type = "flatten2"
+            self.place = paddle.XPUPlace(0)
+            self.init_test_case()
+            self.inputs = {
+                "X": np.random.random(self.in_shape).astype("float32")
```
Comment (XiaociZhang) [inline python/paddle/fluid/tests/unittests/xpu/test_flatten2_op_xpu.py]: 改成self.dtype

File: python/paddle/fluid/tests/unittests/xpu/test_flatten2_op_xpu.py [hunk 2]
```diff
@@ -23,67 +23,77 @@
...
+            }
+            self.init_attrs()
+            self.outputs = {
+                "Out": self.inputs["X"].reshape(self.new_shape),
+                "XShape": np.random.random(self.in_shape).astype("float32")
```
Comment (XiaociZhang) [inline python/paddle/fluid/tests/unittests/xpu/test_flatten2_op_xpu.py]: 同上

File: python/paddle/fluid/tests/unittests/xpu/test_flatten2_op_xpu.py [hunk 3]
```diff
@@ -23,67 +23,77 @@
...
+        self.op_name = 'flatten2'
+        self.use_dynamic_create_class = False
 
-    def set_xpu(self):
-        self.__class__.use_xpu = True
+    class TestFlatten2Op(XPUOpTest):
 
-    def test_check_output(self):
-        self.check_output_with_place(self.place, no_check_set=["XShape"])
+        def setUp(self):
```
Comment (XiaociZhang) [inline python/paddle/fluid/tests/unittests/xpu/test_flatten2_op_xpu.py:39]: self.dtype = self.in_type

**Review by XiaociZhang — APPROVED** [review]
lgtm

**Review by QingshuChen — APPROVED** [review]

**Pull request merged**